pkg/operation/botanist/addons.go

```diff
@@ -504,3 +525,26 @@ func (b *Botanist) generateOptionalAddonsChart(_ context.Context) (*chartrendere
 func (b *Botanist) outOfClusterAPIServerFQDN() string {
 	return fmt.Sprintf("%s.", b.Shoot.ComputeOutOfClusterAPIServerAddress(b.APIServerAddress, true))
 }
+
+// getCoreDNSRestartTimestamp returns a timestamp that can potentially restart the CoreDNS deployment.
+func (b *Botanist) getCoreDNSRestartTimestamp(ctx context.Context) (string, error) {
```

Comment:
I have the idea to restart only when the deployment is running on the maximum allowed replicas. Any opinions?